.github/actions/connector-image-build-push/action.yml

```
@@ -0,0 +1,324 @@
+            IMAGE_EXISTS="true"
+            DO_BUILD="false"
+            DO_PUBLISH="false"
+            ;;
+          "404")
+            echo "✅ Image does not exist on Docker Hub"
+            IMAGE_EXISTS="false"
+            DO_BUILD="true"
+            DO_PUBLISH="true"
+            ;;
+          *)
+            echo "⚠️ Unable to determine if image exists (HTTP $STATUS)... Aborting to be safe"
+            exit 1
+            ;;
+        esac
+
+        # If dry-run is enabled, set build and publish decisions accordingly
+        if [[ "${{ inputs.dry-run }}" == "true" ]]; then
+          echo "🔄 Dry run mode enabled - will build but not push"
+          DO_BUILD="true"
+          DO_PUBLISH="false"
+        fi
+
+        # If force-publish is enabled, override decisions
+        if [[ "${{ inputs.force-publish }}" == "true" ]]; then
+          echo "🔄 Force publish enabled - will build and push regardless of existing image"
+          DO_BUILD="true"
+          DO_PUBLISH="true"
+        fi
+
+        # Output decision summary
+        echo ""
+        echo "📊 Build Decision Summary:"
+        echo "  Image exists: $IMAGE_EXISTS"
+        echo "  Will build: $DO_BUILD"
+        echo "  Will publish: $DO_PUBLISH"
+        echo "  Push latest tag: ${{ inputs.push-latest }}"
+        echo "  Dry run mode: ${{ inputs.dry-run }}"
+        echo "  Force publish: ${{ inputs.force-publish }}"
+
+        # Set outputs for subsequent steps
+        {
+          echo "image-exists=$IMAGE_EXISTS"
+          echo "do-build=$DO_BUILD"
+          echo "do-publish=$DO_PUBLISH"
+        } >> $GITHUB_OUTPUT
+
+    - name: Login to Docker Hub
+      if: ${{ steps.check-exists.outputs.do-publish == 'true' && inputs.docker-hub-username != '' && inputs.docker-hub-password != '' }}
+      uses: docker/login-action@v3
+      with:
+        username: ${{ inputs.docker-hub-username }}
+        password: ${{ inputs.docker-hub-password }}
+
+    - name: Set up Docker Buildx
+      if: ${{ steps.check-exists.outputs.do-build == 'true' }}
+      uses: docker/setup-buildx-action@v3
+
+    - name: Build connector image for testing
+      if: ${{ steps.check-exists.outputs.do-build == 'true' }}
+      uses: docker/build-push-action@v5
+      with:
+        context: ${{ steps.vars.outputs.connector-dir }}
+        file: docker-images/Dockerfile.${{ steps.vars.outputs.connector-type }}-connector
+        platforms: local
+        tags: ${{ steps.vars.outputs.docker-tags }}
+        build-args: |
+          BASE_IMAGE=${{ steps.vars.outputs.base-image }}
+          CONNECTOR_NAME=${{ steps.vars.outputs.connector-name }}
+        load: true
+        push: false
+
+    - name: Run `spec` Image Test
+      if: ${{ steps.check-exists.outputs.do-build == 'true' }}
+      shell: bash
+      run: |
+        image_name="${{ steps.vars.outputs.connector-image-name }}"
+        echo "Running spec test for image: $image_name"
+        docker run --rm $image_name spec
```

```suggestion
        docker run --rm $image_name spec > /dev/null
```

(I don't think we really need to print the `spec` output to the action logs?)